python-frontend/src/test/java/org/sonar/python/tree/TreeUtilsTest.java

```diff
@@ -441,6 +441,49 @@ public void test_getTreeByPositionComparator() {
     assertThat(comparsionResult).isPositive();
   }
 
+  @Test
+  public void test_findIndent() {
+    FileInput fileInput = PythonTestUtils.parse("def foo():\n" +
+      "    if a < 3: pass\n");
+
+    var passDef = PythonTestUtils.getLastDescendant(fileInput, t -> t.is(Kind.PASS_STMT));
+
+    var indent = TreeUtils.findIndent(passDef);
+    assertThat(indent).isEqualTo(4);
+  }
+
+  @Test
+  public void test_findIndentDownTree() {
+    FileInput fileInput = PythonTestUtils.parse("if a < 3: pass\n" +
+      "\n" +
+      "def foo(a):\n" +
+      "  print(a)");
+
+    var passDef = PythonTestUtils.getLastDescendant(fileInput, t -> t.is(Kind.PASS_STMT));
+
+    var indent = TreeUtils.findIndent(passDef);
+    assertThat(indent).isEqualTo(2);
+  }
+
+  @Test
+  public void test_findIndentZero() {
+    FileInput fileInput = PythonTestUtils.parse("if a < 3: pass\n" +
+      "\n" +
+      "def foo(a): pass");
+
+    var passDef = PythonTestUtils.getLastDescendant(fileInput, t -> t.is(Kind.PASS_STMT));
+
+    var indent = TreeUtils.findIndent(passDef);
+    assertThat(indent).isZero();
+  }
+
+  @Test
+  public void test_findIndentEmpty() {
+    FileInput fileInput = PythonTestUtils.parse("");
+    var indent = TreeUtils.findIndent(fileInput);
+    assertThat(indent).isZero();
+  }
+
```

Comment:
Would be nice adding an example with inconsistent indentation, for example in same cases it's 2 and in other cases it's 4

Reply:
done